sale_purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = "purchase.requisition"
+
+    sale_id = fields.Many2one(
+        comodel_name="sale.order",
+        string="Sale Order",
+        help="Sale order related to this purchase requisition.",
+    )
+    sale_user_id = fields.Many2one(
+        comoel_name="res.users",
+        string="Sale Representative",
+        related="sale_id.user_id",
+    )
+    alert_activity = fields.Boolean(default=False)
+    purchase_requisition_sync_warning = fields.Char(compute="_compute_sync_sale_order")
+
+    def _compute_sync_sale_order(self):
+        for rec in self:
+            rec.purchase_requisition_sync_warning = False
+            if rec.sale_id.purchase_requisition_sync_warning:
+                rec.purchase_requisition_sync_warning = _(
+                    "Some lines are not synced on these Sale order:"
+                    + "%s please update this lines manually" % (rec.sale_id.name),
+                )
+                if (
+                    not rec.alert_activity
+                    and len(self) == 1
+                    and (rec.user_id or rec.sale_user_id)
+                ):
```

Comment:
It makes no sense to check `len(self) == 1` because it's always gonna be. In this method's 1st line you're iterating the records already.